core/compressed_translation.cpp

```diff
@@ -46,6 +46,7 @@ struct _PHashTranslationCmp {
 void PHashTranslation::generate(const Ref<Translation> &p_from) {
 #ifdef TOOLS_ENABLED
 	List<StringName> keys;
+	ERR_FAIL_COND(p_from.is_null());
```

Comment:
This should go before `List<StringName> keys;`.

Reply:
Can you explain why should it be done in these way? In my mind it is good solution because while we are reading function we are does not think about things that are does not need us right now(at this line or few later) but in this situation I has been confused because we use this in "few later".

Reply:
It was not "wrong" initially but the code is more readable this way, and avoids needlessly allocating memory in the case where we would trigger an error:

- First we check that the parameter is valid. If not, there's nothing to do so we should return right away.
- If the parameter is valid, then we can proceed with the normal logic: create a `List`, fill it, etc.

Sometimes it does make sense to do an `ERR_FAIL_COND` check before the first use of what needs to be validated before de-referencing, but here an invalid `p_from` means that the whole method is useless, so it's cleaner to return as early as possible.

Sometimes we might have situations like this:

```
void some_func(const Ref<T> &p_arg) {
    // do some very expensive calculations which don't use p_arg, and don't impact the global state.
    Variant local_var = ...;
    ERR_FAIL_NULL(p_arg);
    p_arg->some_method(local_var);
```

If we put the error before `p_arg`, we would calculate the expensive `local_var` stuff needlessly.

Reply:
Okay. Thank you.